app/src/main/java/com/hedvig/app/feature/offer/ui/checkout/CheckoutActivity.kt

```diff
@@ -67,28 +75,84 @@ class CheckoutActivity : BaseActivity(R.layout.activity_checkout) {
             }
         }
 
-        viewModel.viewState
+        viewModel.inputViewState
             .flowWithLifecycle(lifecycle)
             .onEach(::setInputState)
             .launchIn(lifecycleScope)
+
+        viewModel.titleViewState
+            .flowWithLifecycle(lifecycle)
+            .onEach(::setTitleState)
+            .launchIn(lifecycleScope)
+
+        viewModel.events
+            .flowWithLifecycle(lifecycle)
+            .onEach(::handleEvent)
+            .launchIn(lifecycleScope)
     }
 
-    private fun setInputState(viewState: CheckoutViewModel.ViewState) {
+    private fun setInputState(viewState: CheckoutViewModel.InputViewState) {
         binding.signButton.isEnabled = viewState.enableSign
         setContainerInputState(binding.emailInputContainer, viewState.emailInputState)
         setContainerInputState(binding.identityNumberInputContainer, viewState.identityInputState)
     }
 
     private fun setContainerInputState(
         textInputLayout: TextInputLayout,
-        state: CheckoutViewModel.ViewState.InputState
+        state: CheckoutViewModel.InputViewState.InputState
     ) {
         when (state) {
-            is CheckoutViewModel.ViewState.InputState.Invalid -> {
+            is CheckoutViewModel.InputViewState.InputState.Invalid -> {
                 textInputLayout.error = getString(state.stringRes ?: R.string.component_error)
             }
-            CheckoutViewModel.ViewState.InputState.NoInput,
-            is CheckoutViewModel.ViewState.InputState.Valid -> textInputLayout.error = null
+            CheckoutViewModel.InputViewState.InputState.NoInput,
+            is CheckoutViewModel.InputViewState.InputState.Valid -> textInputLayout.error = null
+        }
+    }
+
+    private fun setTitleState(titleState: CheckoutViewModel.TitleViewState) {
+        when (titleState) {
+            is CheckoutViewModel.TitleViewState.Loaded -> {
+                TransitionManager.beginDelayedTransition(binding.root)
+
+                binding.title.show()
+                binding.cost.show()
+                binding.originalCost.isVisible = !titleState.netAmount.subtract(titleState.grossAmount).isZero
```

Comment:
Should we use an `operator fun` here? I think that would read better than `.subtract(other)` 🙂